llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -8430,8 +8442,16 @@ unsigned SIInstrInfo::getLiveRangeSplitOpcode(Register SrcReg,
 }
 
 bool SIInstrInfo::isBasicBlockPrologue(const MachineInstr &MI) const {
-  return !MI.isTerminator() && MI.getOpcode() != AMDGPU::COPY &&
-         MI.modifiesRegister(AMDGPU::EXEC, &RI);
+  // It should cover all possible instructions (spills, copies, etc.) early
+  // inserted during RA in the block prolog to avoid the broken prologs that
```

Comment:
I think the "broken prolog" reads a bit too vaguely. Also not sure what "early" is doing here. How about

"We need to handle instructions which may be inserted during register allocation to handle the prolog. The initial prolog instruction may have been separated from the start of the block by spills and copies inserted needed by the prolog"

I still don't feel particularly comfortable trusting any spills here were for the prolog

Reply:
Done